### PR TITLE
can load next level children.

### PR DESCRIPTION
**Pull request by cyjia**

### CODE REVIEW

File: addon/controllers/grouped-row-array.js [hunk 1]
```diff
@@ -30,7 +30,9 @@ export default RowArrayController.extend({
 
   expandChildren: function(row) {
     row.set('isExpanded', true);
-    var childrenRow = row.get('children') || [];
+    var childrenRow = row.get('children');
+    var self = this;
+    childrenRow.addObserver('length', this, 'lengthDidChange');
```
Comment (jefflage) [inline addon/controllers/grouped-row-array.js]: adding and removing observers is generally considered bad.  is it possible to do this with an action from the view that renders the +/- instead?

File: addon/models/grouping-row-proxy.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  loadChildren: Ember.K,
+
+  children: Ember.computed(function () {
+    var self = this;
+    this.loadChildren().then(function (x) {
+      self.set('_children.content', x);
+    });
+    return this.get('_children');
+  }).property('content')
+});
```
Comment (jefflage) [inline addon/models/grouping-row-proxy.js]: this looks dangerous:
1. its a getter with side effects in that it sets another property
2. derived props in ember dont 'respect' promises/async operations

Comment (cyjia) [inline addon/models/grouping-row-proxy.js]: We want to implicitly trigger the action of loading children.

File: addon/controllers/grouped-row-array.js (hunk 1 above)
Comment (cyjia) [inline addon/controllers/grouped-row-array.js]: We want to implicitly trigger the action of loading children. `ember-table` needs to be notified when more children are loaded. Using observer is a quick and dirty solution.

Comment (jefflage) [inline addon/controllers/grouped-row-array.js]: is this temporary or will you remove it?

File: addon/models/grouping-row-proxy.js (hunk 2 above)
Comment (jefflage) [inline addon/models/grouping-row-proxy.js]: we should look for another solution as this pattern has caused us problems in the past

File: addon/controllers/grouped-row-array.js (hunk 1 above)
Comment (cyjia) [inline addon/controllers/grouped-row-array.js]: Now we want to find a solution without using `addObserver`.

File: addon/models/grouping-row-proxy.js (hunk 2 above)
Comment (cyjia) [inline addon/models/grouping-row-proxy.js]: Is the same pattern here? 
- `children` is a ArrayProxy and is expected to be calculated only once. If `loadChildren` is not finished, its `content' is a placeholder `[{'id': 'placeholder', isLoading: true}]`
- `children.content` will be updated whenever `loadChildren` finished, but the computed property is not expected to be calculated again

**Pull request closed without merging**